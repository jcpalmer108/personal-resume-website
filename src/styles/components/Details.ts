import styled from "styled-components";

export const Wrapper = styled.div`
  padding-top: 35px;
  display: flex;
  flex-direction: column;
  row-gap: 10px;

  &:first-child {
    padding-top: 45px;
  }

`

export const Label = styled.div`
  font-family: Gilroy-Bold;
  font-size: 15px;
`

export const Info = styled.p`
  font-family: Gilroy-Regular;
  font-size: 12px;
  margin: 0px;
`