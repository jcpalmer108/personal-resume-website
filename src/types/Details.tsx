export type WrapperProps = {
  noBottom?: boolean | undefined;
}

export type DetailsProps = {
  label: string,
  info?: string[],
  noBottom?: boolean
}
