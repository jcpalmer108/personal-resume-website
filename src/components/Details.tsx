import { Wrapper, Label, Info } from "../styles/components/Details";
import { DetailsProps } from "../types/Details";

export default function Details({ label, info, noBottom }: DetailsProps) {
  if (!info) {
    return null;
  }

  return (
    <Wrapper noBottom={noBottom} data-testid="Details">
      <Label>{label}</Label>
      <div>
        {info.map((paragraph, index) => (
          <Info key={`Info ${index + 1}`}>{paragraph}</Info>
        ))}
      </div>
    </Wrapper>
  );
}
