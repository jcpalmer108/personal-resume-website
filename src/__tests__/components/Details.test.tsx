import { render, screen } from "@testing-library/react";
import Details from "../../components/Details"

describe('Details', () => {
  const params = {
    label: 'test label',
    info: [ '1', '2' ],
    noBottom: true,
  }

  test('renders if optional and required params are passed in', () => {
    render(<Details label={params.label} info={params.info} noBottom />)
    expect(screen.getByTestId("Details")).toMatchSnapshot()
    expect(screen.getByTestId("Label")).toHaveTextContent(params.label)
    expect(screen.getByTestId("Description")).toHaveTextContent(params.info.join(""))
  })

  test('renders if only required params are passed in', () => {
    render(<Details label={params.label} info={params.info} />)
    expect(screen.getByTestId("Details")).toMatchSnapshot()
    expect(screen.getByTestId("Label")).toHaveTextContent(params.label)
    expect(screen.getByTestId("Description")).toHaveTextContent(params.info.join(""))
  })
})
