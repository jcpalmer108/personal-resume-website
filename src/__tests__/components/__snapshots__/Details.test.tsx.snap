// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details renders if optional and required params are passed in 1`] = `
<div
  class="sc-bczRLJ eVwvdg"
  data-testid="Details"
>
  <div
    class="sc-gsnTZi jqxNpn"
  >
    test label
  </div>
  <div>
    <p
      class="sc-dkzDqf dVromW"
    >
      1
    </p>
    <p
      class="sc-dkzDqf dVromW"
    >
      2
    </p>
  </div>
</div>
`;
